File: Renaissance.AspNet/frontend/Renaissance.Web/Services/NotificationService.cs
namespace Renaissance.Web.Services;

public class NotificationService
{
    public string? Message { get; private set; }

    public void ShowSuccess(string message) => Message = message;

    public string? TakeMessage()
    {
        var message = Message;
        Message = null;
        return message;
    }
}

File: Renaissance.AspNet/frontend/Renaissance.Angular/src/app/core/services/notification.service.ts
import { Injectable, inject } from '@angular/core';
import { MatSnackBar } from '@angular/material/snack-bar';

@Injectable({ providedIn: 'root' })
export class NotificationService {
    private readonly snackBar = inject(MatSnackBar);

    showSuccess(message: string, durationMs = 4000): void {
        this.snackBar.open(message, 'Close', {
            duration: durationMs,
            panelClass: ['snackbar-success']
        });
    }

    showError(message: string, durationMs = 6000): void {
        this.snackBar.open(message, 'Close', {
            duration: durationMs,
            panelClass: ['snackbar-error']
        });
    }
}
